scripts/ecs/appsignals/setup-ecs-demo.sh

```diff
@@ -0,0 +1,1124 @@
+    aws iam attach-role-policy --role-name $IAM_EXECUTION_ROLE_NAME --policy-arn "arn:aws:iam::aws:policy/AmazonECS_FullAccess"
+
+    operation_id=$(aws servicediscovery create-private-dns-namespace \
+      --name ecs-petclinic \
+      --vpc $VPC_ID \
+      --query 'OperationId' \
+      --output text)
+
+    echo "Namespace creation initiated. Waiting for the namespace to be available..."
+    while true; do
+      STATUS=$(aws servicediscovery get-operation \
+        --operation-id $operation_id \
+        --query 'Operation.Status' \
+        --output text)
+
+      if [ "$STATUS" == "SUCCESS" ]; then
+        echo "Namespace is now available!"
+        break
+      elif [ "$STATUS" == "FAIL" ]; then
+        echo "Namespace creation failed."
+        exit 1
+      else
+        echo "Namespace is still being created. Status: $STATUS"
+        sleep 30
+      fi
+    done
+
+    # Creat ECS cluster
+    aws ecs create-cluster --cluster-name ${CLUSTER} > /dev/null
+
+    # Get ECR image prefix
+    ECR_IMAGE_PREFIX=$(aws ecr-public describe-repositories --repository-names traffic-generator --region us-east-1 --query 'repositories[0].repositoryUri' --output text | cut -d'/' -f1,2)
+
+    echo "VPC_ID=\"$VPC_ID\"" >> $OUTPUT_FILE
+    echo "SUBNET_IDS=\"$SUBNET_IDS\"" >> $OUTPUT_FILE
+    echo "SECURITY_GROUP_ID=\"$SECURITY_GROUP_ID\"" >> $OUTPUT_FILE
+    echo "LOAD_BALANCER_ARN=\"$LOAD_BALANCER_ARN\"" >> $OUTPUT_FILE
+    echo "LOAD_BALANCER_DNS=\"$LOAD_BALANCER_DNS\"" >> $OUTPUT_FILE
+    echo "ECR_IMAGE_PREFIX=\"$ECR_IMAGE_PREFIX\"" >> $OUTPUT_FILE
+    echo "ACCOUNT_ID=\"$ACCOUNT_ID\"" >> $OUTPUT_FILE
+
+    # Confirm the output file has been written
+    echo "Variables written to $OUTPUT_FILE"
+
+    echo "Resource creation complete."
+
+}
+
+function run_config_server() {
+  sed -i '' "s|\"config-server-image\"|\"${ECR_IMAGE_PREFIX}/springcommunity/spring-petclinic-config-server\"|" ./sample-app/task-definitions/spring-petclinic-config-server.json
+  sed -i '' "s|us-west-2|${REGION}|g" ./sample-app/task-definitions/spring-petclinic-config-server.json
+  sed -i '' "s|000111222333|${ACCOUNT_ID}|g" ./sample-app/task-definitions/spring-petclinic-config-server.json
+
+  aws ecs register-task-definition --cli-input-json file://sample-app/task-definitions/spring-petclinic-config-server.json > /dev/null
+
+  namespace_id=$(aws servicediscovery list-namespaces --query "Namespaces[?Name=='ecs-petclinic'].Id" --output text)
+  service_discovery_id=$(aws servicediscovery create-service \
+      --name config-server \
+      --dns-config "NamespaceId=$namespace_id,RoutingPolicy=WEIGHTED,DnsRecords=[{Type=A,TTL=300}]" \
+      --health-check-custom-config FailureThreshold=2 \
+      --query "Service.Id" --output text)
+
+  registryArn=$(aws servicediscovery get-service \
+    --id $service_discovery_id \
+    --query 'Service.Arn' --output text)
+
+  aws ecs create-service \
+      --service-name config-server \
```

Comment:
nit: can we introduce a variable into service name so that there won't be name conflicts if creating multiple ecs demo suites in one region?

Reply:
Added `$CLUSTER` into service name, for example: `discovery-server-$CLUSTER.ecs-petclinic`